config.go

```diff
@@ -421,13 +421,18 @@ func marshal(c Config) *bytes.Buffer {
 // Pattern is a pattern in a Host declaration. Patterns are read-only values;
 // create a new one with NewPattern().
 type Pattern struct {
-	str   string // Its appearance in the file, not the value that gets compiled.
-	regex *regexp.Regexp
-	not   bool // True if this is a negated match
+	str           string // Its appearance in the file, not the value that gets compiled.
+	regex         *regexp.Regexp
+	not           bool // True if this is a negated match
+	hasWhitespace bool
 }
 
 // String prints the string representation of the pattern.
 func (p Pattern) String() string {
+	if p.hasWhitespace {
+		return fmt.Sprintf("\"%s\"", p.str)
```

Comment:
should we do `fmt.Sprintf("%q")` here?

Reply:
I don't think so. `String()` must return a representation that would work in an actual SSH config file, but `%q` escapes characters like `"`, `\n`, `\r`, `\t`. And the pattern string:
- cannot contain `"`, because of the logic in `parseSSHArguments`;
- cannot contain `\n`, `\r`, because of how the lexer parses a config;
- can contain `\t` (and probably other characters that `%q` would escape), but they must be left as-is, as are not escaped in an SSH config.